### PR TITLE
Add LIP: "Introduce a generic keystore"

### PR DESCRIPTION
**Pull request by ricott1**

### CODE REVIEW

**Review by mitsuaki-u — APPROVED** [review]

File: proposals/lip-introduce_a_genric_keystore.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+The `version` is set to `"1"`.
+
+##### ciphertext
+
+The encrypted message in hexadecimal format.
+
+##### mac
+
+Computed as `SHA256(last 16 bytes of derivation key || ciphertext)`. It can be used to check the verification key before starting the decryption process.
```
Comment (sergeyshemyakov) [inline proposals/lip-introduce_a_genric_keystore.md:97]: More precisely:
`SHA256(derivationKey[-16:] + cipherText)`

File: proposals/lip-introduce_a_genric_keystore.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+List of paths used with the store recovery phrate to derive key pairs. This property should be used only if the encoded data is a secret recovery phrase. This information is useful to recover all accounts that were generated with this recovery phrase. It is also useful when creating a new account and selecting the next unused path.
+
+##### tags
+
+List of tags associated with the file.
+
+#### id
+
+The `id` property stores a provided uuid (version 4 UUID as specified in [RFC 4122](https://datatracker.ietf.org/doc/html/rfc4122)), this is a randomly generated ID. It is used if the keystore needs to be referred to. Implementation help: the generation of the id is supported by node.js with the [`uuid`](https://github.com/uuidjs/uuid#uuidv4options-buffer-offset) package for example.
```
Comment (sergeyshemyakov) [inline proposals/lip-introduce_a_genric_keystore.md:168]: >id

Capitalize: ID

File: proposals/lip-introduce_a_genric_keystore.md (hunk 1 above)
Comment (ricott1) [reply proposals/lip-introduce_a_genric_keystore.md]: fixed

File: proposals/lip-introduce_a_genric_keystore.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+The encryption/decryption key is an intermediate key derived from the user password. It is used to generate the secret key for decryption, and verify if the given password is correct. The function, and the params used to derive this key from the password are specified in `kdf`. The following values of `kdf` and `kdfparams` are allowed, depending on the key-derivation function:
+
+| `kdf` | function | `kdfparams` | Definition |
+|-------|----------|-------------|------------|
+| "PBKDF2-SHA-256" | `pbkdf2`    | `{iterations: uint32, salt: string}` | [RFC 2898](https://www.ietf.org/rfc/rfc2898.txt)|
+| "argon2id"       | `argon2id`  | `{parallelism: uint32, iterations: uint32, memory: uint32, salt: string}` | [RFC 9106](https://datatracker.ietf.org/doc/html/rfc9106)|
+
+##### cipher and cipherparams
+
+The specified function encrypts the secret using the decryption key; to decrypt it, the decryption key along with `cipher` and `cipherparams` must be used. If the decryption key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
```
Comment (gkoumout) [inline proposals/lip-introduce_a_genric_keystore.md:110]:
```suggestion
The specified function encrypts the secret key using the encryption key; to decrypt it, the decryption key along with `cipher` and `cipherparams` must be used. If the decryption key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
```

File: proposals/lip-introduce_a_genric_keystore.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+The `version` is set to `"1"`.
+
+##### ciphertext
+
+The encrypted message in hexadecimal format.
+
+##### mac
+
+Computed as `SHA256(encryptionKey[-16:] + cipherText)`, where `encryptionKey` is the key derived from the user password using the key-derivation function. It can be used to verify the encryption key before starting the decryption process.
```
Comment (gkoumout) [inline proposals/lip-introduce_a_genric_keystore.md:97]: user password is first mentioned here. Perhaps add a one-sentence explanation (either here or in the rationale) on what the password to make clear that it's different than the secret recovery phrase.

**Review by gkoumout — CHANGES_REQUESTED** [review]
Looks good to me! Just added two short comments!

File: proposals/lip-introduce_a_genric_keystore.md (hunk 3 above)
Comment (ricott1) [reply proposals/lip-introduce_a_genric_keystore.md:110]: I updated several instances in the LIP, to clarify that there is indeed only on key

File: proposals/lip-introduce_a_genric_keystore.md (hunk 4 above)
Comment (ricott1) [reply proposals/lip-introduce_a_genric_keystore.md]: added

**Review by gkoumout — APPROVED** [review]
Looks great!

**Review by karmacoma — APPROVED** [review]

**Pull request merged**